### PR TITLE
Wait for confirmations when polling transactions

### LINKED ISSUE #2664: Unable to start a node because of a missing status field in receipt
## Problem Definition

My node became unusable and it can no longer start because it crashes with the following stacktrace: 

```
Traceback (most recent call last):
  File "/home/cosmin/.virtualenvs/raiden/bin/raiden", line 11, in <module>
    load_entry_point('raiden', 'console_scripts', 'raiden')()
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/__main__.py", line 11, in main
    run(auto_envvar_prefix='RAIDEN')  # pylint: disable=no-value-for-parameter
  File "/home/cosmin/.virtualenvs/raiden/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/cosmin/.virtualenvs/raiden/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/cosmin/.virtualenvs/raiden/lib/python3.6/site-packages/click/core.py", line 1043, in invoke
    return Command.invoke(self, ctx)
  File "/home/cosmin/.virtualenvs/raiden/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/cosmin/.virtualenvs/raiden/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/cosmin/.virtualenvs/raiden/lib/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/ui/cli.py", line 1204, in run
    NodeRunner(kwargs, ctx).run()
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/ui/cli.py", line 1035, in run
    app = self._run_app()
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/ui/cli.py", line 1054, in _run_app
    app_ = run_app(**self._options)
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/ui/cli.py", line 851, in run_app
    raiden_app.start()
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/app.py", line 139, in start
    self.raiden.start()
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/raiden_service.py", line 347, in start
    self.raiden_event_handler.on_raiden_event(self, transaction)
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/raiden_event_handler.py", line 93, in on_raiden_event
    self.handle_contract_send_channelsettle(raiden, event)
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/raiden_event_handler.py", line 449, in handle_contract_send_channelsettle
    partner_locksroot,
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/network/proxies/payment_channel.py", line 214, in settle
    partner_locksroot=partner_locksroot,
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/network/proxies/token_network.py", line 935, in settle
    receipt_or_none = check_transaction_threw(self.client, transaction_hash)
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/network/rpc/transactions.py", line 8, in check_transaction_threw
    if 'status' not in receipt:
TypeError: argument of type 'NoneType' is not iterable
```
I want to mention that I am using infura and the tx_hash is: https://ropsten.etherscan.io/tx/0x0ca6dad843243eb1ea0ec9bba58864fc2c20b8a5d749c3585ab9112b9a0349fe
Full log: 
[log.txt](https://github.com/raiden-network/raiden/files/2445902/log.txt)

### PR DESCRIPTION
**Pull request by palango**
Now RPCClient.poll() waits for DEFAULT_NUMBER_OF_CONFIRMATIONS_BLOCK
confirmations by default. This is done to avoid issues like #2664 

I checked all calls and none overwrites the default.

Part of #2664

### CODE REVIEW

**Comment by LefterisJP** [comment]
Also lgtm. Rebased on master to fix test build.

**Comment by palango** [comment]
Locally with `--blockchain-type=geth` the `integration/contracts` tests pass, so `--blockchain-type=tester` seems to be the culprit here.

**Comment by LefterisJP** [comment]
@palango perhaps it's time? https://github.com/raiden-network/raiden/pull/2738

**Comment by palango** [comment]
This increases test times a lot. We need to set this lower for testing.

**Comment by ulope** [comment]
Maybe this is also a candidate for different values on test/main network?

File: raiden/network/rpc/client.py [hunk 1]
```diff
@@ -446,21 +455,18 @@ def poll(self, transaction_hash: bytes, confirmations: int = None):
...
+                block_number = self.block_number()
 
-            block_number = self.block_number()
+                if block_number >= confirmation_block:
+                    return transaction
 
-            while block_number < confirmation_block:
-                gevent.sleep(.5)
-                block_number = self.block_number()
+            gevent.sleep(1.0)
```
Comment (LefterisJP) [inline raiden/network/rpc/client.py]: Why did you increase the polling timeout to 1 second from 0.5 here?

File: raiden/network/rpc/client.py [hunk 2]
```diff
@@ -148,6 +148,7 @@ def __init__(
             privkey: bytes,
             gas_price_strategy: Callable = rpc_gas_price_strategy,
             nonce_offset: int = 0,
+            block_num_confirmations: int = 0,
```
Comment (LefterisJP) [inline raiden/network/rpc/client.py]: Have you made sure that in all the integration tests the value of 0 is given? i.e. that for all tests this is 0? Except perhaps one test that could test specifically for confirmations?

File: tools/scenario-player/scenario_player/runner.py [hunk 3]
```diff
@@ -134,6 +134,7 @@ def __init__(
             Web3(HTTPProvider(chain_urls[self.chain_name][0])),
             privkey=account.privkey,
             gas_price_strategy=get_gas_price_strategy(settings.get('gas_price', 'fast')),
+            block_num_confirmations=1,
```
Comment (LefterisJP) [inline tools/scenario-player/scenario_player/runner.py]: Why 1 block confirmation and not 0 in the SP?

File: raiden/network/rpc/client.py [hunk 4]
```diff
@@ -424,6 +428,11 @@ def poll(self, transaction_hash: bytes, confirmations: int = None):
                 'transaction_hash must be a 32 byte hash',
             )
 
+        if self.default_block_num_confirmations < 0:
```
Comment (LefterisJP) [inline raiden/network/rpc/client.py]: Why not test this at the `JSONRPCClient` ctor?

File: raiden/network/rpc/client.py (hunk 1 above)
Comment (palango) [reply raiden/network/rpc/client.py]: With average block times of 15s it doesn't make sense to poll every 0.5s. Especially as we now use `getTransaction` and not only `blockNumber` in the polling, this also helps reducing load on the Ethereum node.

Comment (LefterisJP) [reply raiden/network/rpc/client.py]: All right. I don't disagree -- just wanted to see your way thinking. Can you perhaps put this as  a constant variable somewhere. I dislike magic numbers.

File: raiden/network/rpc/client.py [hunk 5]
```diff
@@ -410,7 +416,10 @@ def send_transaction(
             log.debug('send_raw_transaction returned', tx_hash=encode_hex(tx_hash), **log_details)
             return tx_hash
 
-    def poll(self, transaction_hash: bytes, confirmations: int = None):
+    def poll(
+            self,
+            transaction_hash: bytes,
+    ):
```
Comment (palango) [inline raiden/network/rpc/client.py]: There's a docstring for `confirmations` which should be removed.

**Pull request merged**